org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/FhirValidatorHttpService.java

```diff
@@ -0,0 +1,343 @@
+    if (param == null || param.trim().isEmpty()) {
+      return new ArrayList<>();
+    }
+    return Arrays.asList(param.split(","));
+  }
+
+  private IdStatus parseIdStatus(String param) {
+    if (param == null) return IdStatus.OPTIONAL;
+    return IdStatus.valueOf(param.toUpperCase());
+  }
+
+  private boolean parseBooleanParameter(String param, boolean defaultValue) {
+    if (param == null) return defaultValue;
+    return "true".equalsIgnoreCase(param);
+  }
+
+  private BestPracticeWarningLevel parseBestPracticeWarningLevel(String param) {
+    if (param == null) return BestPracticeWarningLevel.Ignore;
+    return BestPracticeWarningLevel.valueOf(param);
+  }
+
+  private CheckDisplayOption parseCheckDisplayOption(String param) {
+    if (param == null) return CheckDisplayOption.Ignore;
+    return CheckDisplayOption.valueOf(param);
+  }
+
+  private OperationOutcome createSuccessOperationOutcome(String message) {
+    OperationOutcome outcome = new OperationOutcome();
+    OperationOutcome.OperationOutcomeIssueComponent issue = outcome.addIssue();
+    issue.setSeverity(OperationOutcome.IssueSeverity.INFORMATION);
+    issue.setCode(OperationOutcome.IssueType.INFORMATIONAL);
+    issue.setDiagnostics(message);
+    return outcome;
+  }
+
+  private OperationOutcome createErrorOperationOutcome(String message) {
+    OperationOutcome outcome = new OperationOutcome();
+    OperationOutcome.OperationOutcomeIssueComponent issue = outcome.addIssue();
+    issue.setSeverity(OperationOutcome.IssueSeverity.ERROR);
+    issue.setCode(OperationOutcome.IssueType.EXCEPTION);
+    issue.setDiagnostics(message);
+    return outcome;
+  }
+
+  private void sendOperationOutcome(HttpExchange exchange, int statusCode, OperationOutcome outcome, String format) throws IOException {
+    try {
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      if ("xml".equalsIgnoreCase(format)) {
+        new XmlParser().compose(baos, outcome);
+        sendResponse(exchange, statusCode, baos.toString(StandardCharsets.UTF_8.name()), "application/fhir+xml");
+      } else {
+        new JsonParser().compose(baos, outcome);
+        sendResponse(exchange, statusCode, baos.toString(StandardCharsets.UTF_8.name()), "application/fhir+json");
+      }
+    } catch (Exception e) {
+      sendResponse(exchange, 500, "Error serializing response: " + e.getMessage(), "text/plain");
+    }
+  }
+
+  private void sendResponse(HttpExchange exchange, int statusCode, String response, String contentType) throws IOException {
+    byte[] responseBytes = response.getBytes(StandardCharsets.UTF_8);
+    exchange.getResponseHeaders().set("Content-Type", contentType);
+    exchange.sendResponseHeaders(statusCode, responseBytes.length);
+    try (OutputStream os = exchange.getResponseBody()) {
+      os.write(responseBytes);
+    }
+  }
+
+  /**
+   * Main method to start the service
+   */
+  public static void main(String[] args) {
```

Comment:
I think this should only be accessed through the ValidatorCli, where there's already logic to produce the ValidatorEngine from a well documented set of Params.